src/accumulate.jl

```diff
@@ -0,0 +1,38 @@
+# Implements the Hillis--Steele algorithm using global memory
+# See algorithm 1 at https://en.wikipedia.org/wiki/Prefix_sum#Parallel_algorithm
+
+function Base.accumulate!(op::Function, vout::CuVector{T}, v::CuVector) where T
+    vin = T.(v)  # convert to type T
+
+    Δ = 1   # Δ = 2^d
+    n = ceil(Int, log2(length(v)))
+
+    num_threads = 256
+    num_blocks = ceil(Int, length(v) / num_threads)
+
+    for d in 0:n   # passes through data
+        @cuda blocks=num_blocks threads=num_threads _partial_accumulate!(op, vout, vin, Δ)
+
+        vin, vout = vout, vin
+        Δ *= 2
+    end
+
+    return vin
+end
+
+
+function _partial_accumulate!(op, vout, vin, Δ)
```

Comment:
`op` is unused here.

Reply:
Thanks for finding that bug in the code!